src/groups/mqb/mqbblp/mqbblp_queuehandle.cpp

```diff
@@ -1252,16 +1262,7 @@ bsls::Types::Int64 QueueHandle::countUnconfirmed(unsigned int subQueueId) const
 
     bsls::Types::Int64 result = 0;
     if (subQueueId == bmqp::QueueId::k_UNASSIGNED_SUBQUEUE_ID) {
-        for (Downstreams::const_iterator itStream = d_downstreams.begin();
-             itStream != d_downstreams.end();
-             ++itStream) {
-            const bsl::shared_ptr<Downstream>& downstream = (*itStream);
-            if (downstream) {
-                if (downstream->d_data) {
-                    result += downstream->d_data->size();
-                }
-            }
-        }
```

Comment:
Looking at the code, I am not sure now if this loop is actually slow.
It seems that we are not blocked here by any IO, and not blocked by the need to acquire mutexes or smth like this.
Can you estimate how many times we might call this `countUnconfirmed` and how many `d_downstream`s we might have on average?

Suppose we have 100k subqueues, and each queue has 10 downstreams. Then we'll have just 1kk additions here in total across all operation chain links, which doesn't look too bad.

My concern here, what if we made our normal workflow slower by a few ticks when we make these counter updates when it's not really necessary.

Reply:
I think Reliable Broadcast shutdown is going to suffer, up to the point of crushing, as we have it in some cases with overloaded proxies.  Consider thread switching as well.  We replace that at the cost of 2-3 atomic addition per PUSH/CONFIRM.  That cost is small.